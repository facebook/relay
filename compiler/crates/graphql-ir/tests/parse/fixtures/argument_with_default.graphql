mutation ArgumentWithDefault {
  feedbackUnLike(input: {feedbackId: "123"}) {
    feedback {
      body {
        text
      }
    }
  }
}