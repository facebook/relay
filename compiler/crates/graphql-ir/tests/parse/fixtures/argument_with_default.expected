==================================== INPUT ====================================
mutation ArgumentWithDefault {
  feedbackUnLike(input: {feedbackId: "123"}) {
    feedback {
      body {
        text
      }
    }
  }
}
==================================== OUTPUT ===================================
[
    Operation(
        OperationDefinition {
            kind: Mutation,
            name: WithLocation {
                location: argument_with_default.graphql:9:28,
                item: OperationDefinitionName(
                    "ArgumentWithDefault",
                ),
            },
            type_: Object(7),
            variable_definitions: [],
            directives: [],
            selections: [
                LinkedField {
                    alias: None,
                    definition: WithLocation {
                        location: argument_with_default.graphql:33:47,
                        item: FieldID(41),
                    },
                    arguments: [
                        Argument {
                            name: WithLocation {
                                location: argument_with_default.graphql:48:53,
                                item: ArgumentName(
                                    "input",
                                ),
                            },
                            value: WithLocation {
                                location: argument_with_default.graphql:55:74,
                                item: Constant(
                                    Object(
                                        [
                                            ConstantArgument {
                                                name: WithLocation {
                                                    location: argument_with_default.graphql:56:66,
                                                    item: ArgumentName(
                                                        "feedbackId",
                                                    ),
                                                },
                                                value: WithLocation {
                                                    location: argument_with_default.graphql:68:73,
                                                    item: String(
                                                        "123",
                                                    ),
                                                },
                                            },
                                        ],
                                    ),
                                ),
                            },
                        },
                    ],
                    directives: [],
                    selections: [
                        LinkedField {
                            alias: None,
                            definition: WithLocation {
                                location: argument_with_default.graphql:82:90,
                                item: FieldID(168),
                            },
                            arguments: [],
                            directives: [],
                            selections: [
                                LinkedField {
                                    alias: None,
                                    definition: WithLocation {
                                        location: argument_with_default.graphql:99:103,
                                        item: FieldID(138),
                                    },
                                    arguments: [],
                                    directives: [],
                                    selections: [
                                        ScalarField {
                                            alias: None,
                                            definition: WithLocation {
                                                location: argument_with_default.graphql:114:118,
                                                item: FieldID(409),
                                            },
                                            arguments: [],
                                            directives: [],
                                        },
                                    ],
                                },
                            ],
                        },
                    ],
                },
            ],
        },
    ),
]
